### DISCUSSION TITLE
i am getting error "java.security.InvalidKeyException: IOException : algid parse error, not a sequence"

### DISCUSSION THREAD

**Discussion by NeerajGahlawatJava** [openssl/openssl#25362] [Q&A]
i am getting error "java.security.InvalidKeyException: IOException : algid parse error, not a sequence" while Parsing DER format key.

**Comment by nhorman** [comment]
What makes you think this is an openssl error?

**Comment by NeerajGahlawatJava** [comment]
I am not saying it's a openssl error . it is just because of curiosity that what this algid is which is causing issue and restricting to convert DER format private key into java private key instance .

**Reply by nhorman** [reply]
Its not an algid thats causing the error.  Its the fact that the algid field is supposed to be typed as a sequence (DER data type 30), but in your DER file the data in that position is something other than data type 30 - i.e. your input DER file is most likely malformed.  Beyond that, its impossible to say.
